frontend/app/src/utils/zod-parser.ts

```diff
@@ -0,0 +1,12 @@
+import { ZodTypeAny } from 'zod';
+import { logger } from '@/utils/logging';
+
+export function zodParser(parser: ZodTypeAny, rawData: any) {
+  try {
+    return parser.parse(rawData);
+  } catch (e: any) {
+    const err = new Error(e);
+    logger.error('zod failed to parse: ', err.stack);
+    return undefined;
```

Comment:
Is the purpose of this function to just log the parsing errors in the logger, instead of leaving it to the consumer to do so?

The signature of the type is always `Promise<T>` not `Promise<T | undefined>` so all the consumers expect that when a promise completes there will be a value that will not be undefined.

This means that there is a possibility that some parts of the code will try to access the properties of an undefined value, which in turn may also be logged.

At least at the moment, we should avoid doing that and instead keep the previous approach and add the logging at the consumer.

Reply:
Isn't it is the same if I put it in consumer since I still need to do try catch there? @kelsos

Reply:
I see, for the moment I think I'll just keep previous approach. The `result type` mechanism can be good consideration in future.